### ISSUE TITLE
Server Crashing not gracefully stopping

### ISSUE THREAD

**Issue by VindicoRory** [timvisee/lazymc#26]
I am running multiple servers on Ubuntu 22.04 and every single one (from 1.12.2 -> 1.19) seems to crash with error code 143, rather than gracefully shutting down. I am worried that this is going to cause data loss, and it cant be good for the saves.
I have also enabled RCON to see if that resolves any issues but it seems to be the exact same.
The error is shown below:
```shell
WARN  lazymc          > Server process stopped with error code (exit status: 143)
INFO  lazymc::monitor > Server is now sleeping```

**Comment by timvisee** [comment]
The Minecraft server itself crashes, `lazymc` only forwards the error.

Does the server itself run fine without using `lazymc`?

**Comment by obj-obj** [comment]
My server shuts down normally but still has that exact same exit status, maybe the server changes the exit status if there are any warnings or something?

**Comment by marissa999** [comment]
Ok I am pretty sure that the SIGTERM is the issue, not the SIGINT. I just did a small experiment, without lazymc involved:

1. I started my minecraft server in shell A.
`java -Xmx2048m -Xms512m -jar paper-1.19.3-404.jar`
2. Once the server was fully started in shell B I used ps aux to get the PID of the mc server process. 
3. I then executed `kill -2 <PID>` to stop the server with a SIGINT
4. In shell A i saw the server shut down normally.
5. In shell A I then executed `echo $?` to get the return code of the last process in that shell, which was the minecraft server. It printed 0.

I repeated the steps, but at step 3 instead of `kill -2` I used `kill -15` (SIGTERM). At step 5 after I executed the `echo $?` I then got the code 143, which is the same that lazymc returned.

It seems like it matters to the Minecraft server (or jvm?) if a SIGTERM or a SIGINT is used. I guess I could try to write a bash wrapper script, which once it received a sigterm it will forward a sigint.

**Comment by timvisee** [comment]
Thanks for testing this.

When I originally developed this I figured SIGTERM was the best, based documentation. But, maybe this is wrong. Minecraft's documentation is a mess and severely lacking. 

@marissa999 with your server(s), does using SIGINT always exit gracefully?

Note that if stopping fails, a SIGKILL will be sent to force quit the server process to prevent it from getting stuck. I don't think that is relevant here though.

**Comment by marissa999** [comment]
In my testing yesterday, a SIGINT always exit normally, with 0 and no errors. I did no testing with other minecraft server versions, but the one I mentioned.

I also just looked a bit more into this and for some reason there seems like not a lot is documented about this. I did find the following links though:

https://bugs.mojang.com/browse/MC-109811

https://serverfault.com/questions/695849/services-remain-in-failed-state-after-stopped-with-systemctl

So it seems like 143 is working by design? I have not experience with Java, JVM, Minecraft, or Exit codes though. In my opinion I would not expect a 143 from the JVM, even if it stopped gracefully. Hmm. If sending a SIGINT instead of a SIGTERM is the correct approach is something I do not know. Maybe a 143 Error could be ignored? I have no clue what the best thing to do would be.
